puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -284,22 +284,6 @@ def synthesize_edges(hash, catalog)
                 resource_hash = {'type' => resource['type'], 'title' => resource['title']}
                 other_hash = resource_ref_to_hash(other_ref)
 
-                # Puppet doesn't always seem to check this correctly. If we don't
-                # users will later get an invalid relationship error instead.
-                #
-                # Primarily we are trying to catch the non-capitalized resourceref
-                # case problem here: http://projects.puppetlabs.com/issues/19474
-                # Once that problem is solved and older versions of Puppet that have
-                # the bug are no longer supported we can probably remove this code.
-                unless other_ref =~ /^[A-Z][a-z0-9_-]*(::[A-Z][a-z0-9_-]*)*\[.*\]/m
-                  rel = edge_to_s(resource_hash_to_ref(resource_hash), other_ref, param)
-                  raise Puppet::Error, "Invalid relationship: #{rel}, because " +
-                    "#{other_ref} doesn't seem to be in the correct format. " +
-                    "Resource references should be formatted as: " +
-                    "Classname['title'] or Modulename::Classname['title'] (take " +
-                    "careful note of the capitalization)."
-                end
```

Comment:
Puppet 4 does this validation before we ever get here, and is more strict about its validation, which is why we can just delete it here.